File: J_se_les12/HW/Core/Logic/Show/ShowBase.java
package Java.J_se_les12.HW.Core.Logic.Show;

import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutput;
import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public abstract class ShowBase {
    Phonebook phonebook;
    IConsoleOutput output;

    protected ShowBase(Phonebook phonebook, IConsoleOutput output) {
        this.output = output;
        this.phonebook = phonebook;
    }

    public abstract void showContact(int index);
}

File: J_se_les13/HW/Core/Logic/Show/ShowBase.cs
using HW.Core.Phonebook;
using HW.Core.Logic.Interfaces;
namespace HW.Core.Logic.Show
{
    public abstract class ShowBase
    {
        public PhonebookV1 Phonebook { get; set; }

        public IConsoleOutput Output { get; set; }
        protected ShowBase(PhonebookV1 phonebook, IConsoleOutput output)
        {
            Phonebook = phonebook;
            Output = output;
        }
        public abstract void ShowContact(int index);
    }
}